app/src/main/java/co/nimblehq/compose/crypto/ui/navigation/ComposeCryptoApp.kt

```diff
@@ -0,0 +1,45 @@
+package co.nimblehq.compose.crypto.ui.navigation
+
+import android.widget.Toast
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.platform.LocalContext
+import androidx.navigation.NavHostController
+import androidx.navigation.compose.rememberNavController
+import co.nimblehq.compose.crypto.CryptoAppState
+import co.nimblehq.compose.crypto.extension.collectAsEffect
+
+@Composable
+fun ComposeCryptoApp(
+    navController: NavHostController = rememberNavController(),
+    cryptoAppState: CryptoAppState
+) {
+    var onInternetRestore: () -> Unit = {}
+    val context = LocalContext.current
+
+    cryptoAppState.isNetworkConnected.collectAsEffect { isNetworkConnected ->
+        if (isNetworkConnected == false) {
+            val destination = AppDestination.NoNetwork
+
+            val currentRoute = navController.currentBackStackEntry?.destination?.route
+            if (currentRoute == AppDestination.NoNetwork.route) {
+                navController.popBackStack()
+            }
+
+            navController.navigate(destination)
+        }
+    }
+
+    cryptoAppState.isNetworkError.collectAsEffect { error ->
+        Toast.makeText(context, error?.message, Toast.LENGTH_SHORT).show()
+    }
+
+    AppNavigation(
+        navController = navController,
+        onCallBackChange = {
+            onInternetRestore = it
```

Comment:
```suggestion
            callback = it
```

Reply:
Resolved in https://github.com/nimblehq/jetpack-compose-crypto/pull/105/commits/e38967c8b34e0a99c03b3be29b6db05b689ce37e